### PR TITLE
Au revoir CMake 2.0.0

### PR DESCRIPTION
**Pull request by tytan652**
### Description

Remove CMake legacy code path

Except submodules and legacy_check macro since the former depend on it.

PR for submodules and then the macro removal will be done afterward

Note: Ignoring whitespaces is recommended for reviewing "additions".

### Motivation and Context

This path must disapear…

### How Has This Been Tested?

Still build on my machine, let's see for CI.

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,145 +1,35 @@
...
-if(ENABLE_UNIT_TESTS)
-  enable_testing()
-endif()
+add_subdirectory(UI)
 
-if(BUILD_TESTS OR ENABLE_UNIT_TESTS)
-  add_subdirectory(test)
-endif()
+message_configuration()
+return()
```
Comment (RytoEX) [inline CMakeLists.txt:35]: This file looks fine except this `return()`. As this is at the end of the main `CMakeLists.txt` file, it seems that this should not be necessary. @PatTheMav ?

File: plugins/CMakeLists.txt [hunk 2]
```diff
@@ -2,186 +2,91 @@ cmake_minimum_required(VERSION 3.22...3.25)
...
+add_obs_plugin(oss-audio PLATFORMS FREEBSD OPENBSD)
+add_obs_plugin(rtmp-services)
+add_obs_plugin(sndio PLATFORMS LINUX FREEBSD OPENBSD)
+add_obs_plugin(text-freetype2)
+add_obs_plugin(vlc-video WITH_MESSAGE)
+add_obs_plugin(win-capture PLATFORMS WINDOWS)
+add_obs_plugin(win-dshow PLATFORMS WINDOWS)
+add_obs_plugin(win-wasapi PLATFORMS WINDOWS)
+
+return()
```
Comment (RytoEX) [inline plugins/CMakeLists.txt:92]: As with the other review note, I'm not sure that `return()` is necessary now that this is not inside an `if()`, but @PatTheMav would likely know for sure.

**Review by RytoEX — COMMENTED** [review]
The changes look sensible at a visual review, though I'll have to do a test build and consult with @PatTheMav .

File: CMakeLists.txt (hunk 1 above)
Comment (tytan652) [reply CMakeLists.txt]: I forgot to remove them.

**Review by RytoEX — APPROVED** [review]
LGTM

Did a test build and test run on Windows.

**Pull request merged**